go/ql/lib/semmle/go/dataflow/internal/DataFlowNodes.qll

```diff
@@ -11,7 +11,54 @@ private newtype TNode =
   MkSsaNode(SsaDefinition ssa) or
   MkGlobalFunctionNode(Function f) or
   MkImplicitVarargsSlice(CallExpr c) { c.hasImplicitVarargs() } or
-  MkFlowSummaryNode(FlowSummaryImpl::Private::SummaryNode sn)
+  MkFlowSummaryNode(FlowSummaryImpl::Private::SummaryNode sn) or
+  MkPostUpdateNode(IR::Instruction insn) { insn = updatedInstruction() }
+
+private IR::Instruction getADirectlyWrittenInstruction() {
+  exists(IR::WriteTarget target, IR::Instruction base |
+    target = any(IR::WriteInstruction w).getLhs() and
+    (base = result or base = IR::implicitDerefInstruction(result.(IR::EvalInstruction).getExpr()))
+  |
+    target.(IR::FieldTarget).getBase() = base or
+    target.(IR::ElementTarget).getBase() = base
+  )
+  or
+  result = IR::evalExprInstruction(any(SendStmt s).getChannel())
+}
+
+private IR::Instruction getAccessPathPredecessor2(IR::Instruction insn) {
```

Comment:
Looks like getAWrittenNode is now unused, so we could remove that and getAccessPathPredecessor then remove the `2` here